### DISCUSSION TITLE
Handle result of dynamic form

### DISCUSSION THREAD

**Discussion by gourk** [sqlpage/SQLPage#38] [Q&A] [answered]
Hi,

Thank you very much for this project. It allows me to create a small app for a non-profit organisation with a minimal budget.

I have a special problem I cannot deal with, maybe someone here has already faced it.

It is pretty simple to explain:

I have a form defined like that

```
SELECT 'form' AS component,
        './insert.sql' as action,
        'Veuillez passez votre commande' AS title,
        'Commander' AS validate;

SELECT ProductId as name,
        ProductName as label,
        'number' as type,
        0 as min,
        12 as max,
        1 as step
FROM Product;
```

As the input name is from db, I don't know how to retrieve it to build my INSERT query in the insert.sql

Is there a way to query the POST parameters received?

**Comment by lovasoa** [comment, marked as answer]
That's a good question, and that makes me realize we don't have a good documentation for dynamic fields.

I added an example to the repo for precisely this use case: https://github.com/lovasoa/SQLpage/tree/main/examples/forms%20with%20a%20variable%20number%20of%20fields

Let me know if it answers your question.

**Reply by gourk** [reply]
Fantastic ! Thank you very much. You are a star ;-)

**Comment by lovasoa** [comment]
As a note if someone stumbles on this thread now: SQLPage v0.15 will introduce a new [`sqlpage.variables()`](https://sql.ophir.dev/functions.sql?function=variables#function) function, which makes handling this even easier:

```sql
select 'form' as component, 'handle_survey_answer.sql' as action;
select question_id as name, question_text as label from survey_questions;
```

and then handling the results:

```sql
insert into survey_answers(question_id, answer)
select "key", "value" from json_each(sqlpage.variables('post'))
```

Of course, it is still possible to use the square brackets syntax.

I would recommend using:
 - ```sql
    select 'field_name[]' as name
    ```
    and
    ```sql
    insert into my_table select value from json_each(:field_name)
    ```
    when dealing with a simple list, such as a repeated form field where the user enters multiple values for a single question.
 -  [`sqlpage.variables()`](https://sql.ophir.dev/functions.sql?function=variables#function) when dealing with an association between keys and values, such as handling handling results from a form that was generated dynamically from a database table.
